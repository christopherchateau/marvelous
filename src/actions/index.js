export const storeCharacter = (character) => ({
  type: 'STORE_CHARACTER',
  character
})

export const updateStorageDetails = (currentIndex, length) => ({
  });
