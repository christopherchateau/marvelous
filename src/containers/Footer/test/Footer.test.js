import React from "react";
import { shallow } from "enzyme";
import { showFavorites } from "../../../actions";
import { Footer, mapDispatchToProps, mapStateToProps } from "../index";

describe("Footer", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Footer
        dispatchShowFavorites={jest.fn()}
        showFavorites={false}
        path={"/characters"}
      />
    );
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("button text should read 'Favorites' show showFavorites is false", () => {
    const message = wrapper.find(".my-favs");
    expect(message.text()).toBe("Favorites");
  });

  it("button text should read 'Characters' show showFavorites is true", () => {
    wrapper = shallow(
      <Footer
        dispatchShowFavorites={jest.fn()}
        showFavorites={true}
        path={"/favorites"}
      />
    );
    const message = wrapper.find(".my-favs");
    expect(message.text()).toBe("Characters");
  });

  describe("mapDispatchToProps", () => {
    it("should dispatch showFavorites when dispatchShowFavorites is called", () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = showFavorites();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.dispatchShowFavorites();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

  describe("mapStateToProps", () => {
    it("should toggle showFavorites boolean", () => {
      const mockState = { showFavorites: false };
      const expected = { showFavorites: false };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});
