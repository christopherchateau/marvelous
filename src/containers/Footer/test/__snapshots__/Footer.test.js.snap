// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    dispatchShowFavorites={[MockFunction]}
    path="/characters"
    showFavorites={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <NavLink
        activeClassName="active"
        aria-current="page"
        className="fav-btn"
        onClick={[MockFunction]}
        to="/favorites"
      >
        <h3
          className="my-favs"
        >
          Favorites
        </h3>
      </NavLink>,
      "className": "Footer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": <h3
          className="my-favs"
        >
          Favorites
        </h3>,
        "className": "fav-btn",
        "onClick": [MockFunction],
        "to": "/favorites",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorites",
          "className": "my-favs",
        },
        "ref": null,
        "rendered": "Favorites",
        "type": "h3",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <NavLink
          activeClassName="active"
          aria-current="page"
          className="fav-btn"
          onClick={[MockFunction]}
          to="/favorites"
        >
          <h3
            className="my-favs"
          >
            Favorites
          </h3>
        </NavLink>,
        "className": "Footer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <h3
            className="my-favs"
          >
            Favorites
          </h3>,
          "className": "fav-btn",
          "onClick": [MockFunction],
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorites",
            "className": "my-favs",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": "h3",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
